sympy/interactive/printing.py

```diff
@@ -134,20 +135,15 @@ def _matplotlib_wrapper(o):
             return None
 
 
-    from sympy import Basic
-    from sympy.matrices import MatrixBase
-    from sympy.physics.vector import Vector, Dyadic
-    from sympy.tensor.array import NDimArray
+    # Hook methods for builtin sympy printers
+    printing_hooks = ('_latex', '_sympystr', '_pretty', '_sympyrepr')
 
-    # These should all have _repr_latex_ and _repr_latex_orig. If you update
-    # this also update printable_types below.
-    sympy_latex_types = (Basic, MatrixBase, Vector, Dyadic, NDimArray)
```

Comment:
Hmm, this caused a regression in ipython plaintext printing of `MatrixBase`. The summary is:

* previously, we had an explicit pretty printer registered directly for `MatrixBase`
* now that we only register one for the base class, a printer derived from `MatrixBase.__repr__` is used instead of the pretty printer.

Reply:
Fixed in #19902 for `MatrixBase`